src/autoscaler/cf/client.go

```diff
@@ -92,9 +95,24 @@ func NewCFClient(conf *CFConfig, logger lager.Logger, clk clock.Clock) CFClient
 
 	c.lock = &sync.Mutex{}
 
+	uaaClient, err := uaago.NewClient(conf.UAAEndpoint)
+	if err != nil {
+		c.logger.Error("Failed to create uaago client:", err)
+	}
+	c.uaaClient = uaaClient
+
 	return c
 }
 
+func (c *cfClient) GetAuthToken() (string, error) {
```

Comment:
It seems that the function is not called by anyone. Unnecessary?

Reply:
@cdlliuy Its being used during reading from firehose in metricscollector. 

https://github.com/cloudfoundry-incubator/app-autoscaler/pull/362/files#diff-82c5b923bf774ec1cce41431344fe4ceR103

Reply:
Removed the function from here and will put it on PR #362